core/string/to_c_spec.rb

```diff
@@ -38,4 +38,16 @@
       '79+4i'.encode("UTF-16").to_c
     }.should raise_error(Encoding::CompatibilityError, "ASCII incompatible encoding: UTF-16")
   end
+
+  ruby_version_is "3.2" do
+    it "treat a sequence of underscores as an end of Complex string" do
+      "5+3_1i".to_c.should == Complex(5, 31)
+      "5+3__1i".to_c.should == Complex(5)
+      "5+3___1i".to_c.should == Complex(5)
+
+      "12_3".to_c.should == Complex(123)
+      "12__3".to_c.should == Complex(12)
+      "12___3".to_c.should == Complex(12)
+    end
+  end
```

Comment:
**suggestion:** It would be useful also to add similar checks for `Kernel#Complex` method, that is stricter and raises exception when meets multiple `_`. (in a section `"when passed [String]"`)

Reply:
Ah, right. I've overlooked this test case.